src/data.py

```diff
@@ -39,10 +40,10 @@ def set_data_properties(self, parent, import_settings):
         self.unselected_markers = parent.preferences.config["plot_unselected_markers"]
         self.selected_marker_size = parent.preferences.config["plot_selected_marker_size"]
         self.unselected_marker_size = parent.preferences.config["plot_unselected_marker_size"]        
-        if import_settings["name"] != "" and import_settings["mode"] == "single":
-            filename = import_settings["name"]
+        if import_settings.name != "" and import_settings.mode == "single":
+            filename = import_settings.name
         else:
-            filename = import_settings["path"].split("/")[-1]
+            filename = import_settings.path.split("/")[-1]
```

Comment:
Have to say, this syntax looks so much cleaner as object attributes. Was unsure about this change when I read the post, but I think I prefer this.